packages/config/src/browser.ts

```diff
@@ -21,10 +29,12 @@ const debug = Debug('cypress:config:browser')
 const dashesOrUnderscoresRe = /^(_-)+/
 
 // takes an array and creates an index object of [keyKey]: [valueKey]
-function createIndex<T extends Record<string, any>> (arr: Array<T>, keyKey: keyof T, valueKey: keyof T) {
+function createIndex<T extends Record<string, any>> (arr: Array<T>, keyKey: keyof T, valueKey: keyof T, defaultValueFallback?: any) {
   return _.reduce(arr, (memo: Record<string, any>, item) => {
     if (item[valueKey] !== undefined) {
       memo[item[keyKey] as string] = item[valueKey]
+    } else {
+      memo[item[keyKey] as string] = defaultValueFallback
```

Comment:
Seems you don't need the `as string` cast (just tried it, editor seems happy).

Reply:
[cypress-io/cypress@`191105b` (#23040)](https://github.com/cypress-io/cypress/pull/23040/commits/191105ba94ac63d4e1ed69e02bc83d3beab9ec12)